### ISSUE TITLE
Boolean array type results in "newData.isBoolean() && newData.isString()"

### ISSUE THREAD

**Issue by scolyer** [firebase/bolt#89]
I have a type:
```
type Class {
  className: String,
  school: String | Null,
  students: Boolean[]
}
```
which results in:
```
"students": {
          "$key1": {
            ".validate": "newData.isBoolean() && newData.isString()",
            ".read": "auth.uid == $uid",
            ".write": "auth.uid == $uid"
          }
        }
```
But I end up modifying my validate rule to just be `newData.isBoolean()` otherwise I can't write.
Am doing something wrong?

**Comment by mckoss** [comment]
That's a bug.

**Comment by mckoss** [comment]
@scolyer - Could you send me a reproducible test case?

**Comment by mckoss** [comment]
Note: there is still a bug here even if there is an error in the Bolt file;  Bolt should generate an error when two rules create conflicting constraints for the same path.

**Comment by scolyer** [comment]
@mckoss here you go: https://gist.github.com/scolyer/b1116348360ca3644bd3#file-sample-bolt

**Comment by mckoss** [comment]
I think this was fixed by #167 - let me know if you still see problems.
